File: LeetCode/137SingleNumberII/SingleNumberII.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        // Bit Manipulation -> Time:O(n), Space:O(1)
        int one = 0, two = 0;
        for (int i = 0; i < nums.size(); i++) {
            one ^= nums[i];
            one &= ~two;
            two ^= nums[i];
            two &= ~one;
        }
        return one;
    }
};

File: LeetCode/137SingleNumberII/SingleNumberII.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # Bit Manipulation -> Time:O(n), Space:O(1)
        one = 0
        two = 0
        for i in nums:
            one ^= i
            one &= ~two
            two ^= i
            two &= ~one
        return one
